documentation/proposals/Proposal - Generic Math.md

```diff
@@ -0,0 +1,1189 @@
+    [DataMember]
+    public readonly T Y { get; }
+
+    /// <summary>The Z value of the vector component of the quaternion.</summary>
+    [DataMember]
+    public readonly T Z { get; }
+
+    /// <summary>The rotation component of the quaternion.</summary>
+    [DataMember]
+    public readonly T W { get; }
+    
+    public ReadOnlySpan<T> Components { get; }
+
+    /// <summary>Constructs a quaternion from the specified components.</summary>
+    /// <param name="x">The value to assign to the X component of the quaternion.</param>
+    /// <param name="y">The value to assign to the Y component of the quaternion.</param>
+    /// <param name="z">The value to assign to the Z component of the quaternion.</param>
+    /// <param name="w">The value to assign to the W component of the quaternion.</param>
+    public Quaternion(T x, T y, T z, T w);
+
+    /// <summary>Creates a quaternion from the specified vector and rotation parts.</summary>
+    /// <param name="vectorPart">The vector part of the quaternion.</param>
+    /// <param name="scalarPart">The rotation part of the quaternion.</param>
+    public Quaternion(Vector3D<T> vectorPart, T scalarPart);
+
+    /// <summary>Gets a quaternion that represents a zero.</summary>
+    /// <value>A quaternion whose values are <c>(0, 0, 0, 0)</c>.</value>
+    public static Quaternion<T> Zero { get; }
+
+    /// <summary>Gets a quaternion that represents no rotation.</summary>
+    /// <value>A quaternion whose values are <c>(0, 0, 0, 1)</c>.</value>
+    public static Quaternion<T> Identity { get; }
+
+    /// <summary>Gets or sets the element at the specified index.</summary>
+    /// <param name="index">The index of the element to get or set.</param>
+    /// <returns>The element at <paramref name="index" />.</returns>
+    /// <exception cref="ArgumentOutOfRangeException"><paramref name="index" /> was less than zero or greater than the number of elements.</exception>
+    public T this[int index];
+
+    /// <summary>Gets a value that indicates whether the current instance is the identity quaternion.</summary>
+    /// <value><see langword="true" /> if the current instance is the identity quaternion; otherwise, <see langword="false" />.</value>
+    /// <altmember cref="Identity"/>
+    public bool IsIdentity { get; }
+
+    /// <summary>Adds each element in one quaternion with its corresponding element in a second quaternion.</summary>
+    /// <param name="value1">The first quaternion.</param>
+    /// <param name="value2">The second quaternion.</param>
+    /// <returns>The quaternion that contains the summed values of <paramref name="value1" /> and <paramref name="value2" />.</returns>
+    /// <remarks>The <see cref="op_Addition" /> method defines the operation of the addition operator for <see cref="Quaternion" /> objects.</remarks>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Quaternion<T> operator +(Quaternion<T> value1, Quaternion<T> value2);
```

Comment:
nit: Prefer `left` and `right` over `value1` and `value2` for these, since they are clearly the left and right sides of the `+` operator.